pkg/targetgroupbinding/targets_manager.go

```diff
@@ -296,3 +313,27 @@ func pointerizeTargetDescriptions(targets []elbv2sdk.TargetDescription) []*elbv2
 func cloneTargetInfoSlice(targets []TargetInfo) []TargetInfo {
 	return append(targets[:0:0], targets...)
 }
+
+func (m *cachedTargetsManager) isInCIDRRange(ipAddr string) (bool, error) {
+	networks := make([]netip.Prefix, len(m.cidrBlocks))
+	for i, cidr := range m.cidrBlocks {
+		network, err := netip.ParsePrefix(cidr)
+		if err != nil {
+			return false, fmt.Errorf("parse cidr prefix: %w", err)
+		}
+		networks[i] = network
+	}
+
+	ip, err := netip.ParseAddr(ipAddr)
+	if err != nil {
+		return false, fmt.Errorf("parse ip address: %w", err)
+	}
+
+	for _, network := range networks {
+		if network.Contains(ip) {
+			return true, nil
+		}
+	}
```

Comment:
waittt

Reply:
you don't need two loops here

Reply:
It shouldn't be too bad unless we have a million targets;